test/jdk/javax/swing/plaf/basic/BasicPasswordFieldUI/PasswordSelectionWordTest.java

```
@@ -0,0 +1,95 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @key headful
+ * @bug 4231444 8354646
+ * @summary Password fields' ActionMap needs to replace
+ *          DefaultEditorKit.selectWordAction with
+ *          DefaultEditorKit.selectLineAction.
+ *
+ * @run main PasswordSelectionWordTest
+ */
+
+import javax.swing.Action;
+import javax.swing.JPasswordField;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.plaf.basic.BasicTextUI;
+import javax.swing.text.DefaultEditorKit;
+import java.awt.event.ActionEvent;
+
+public class PasswordSelectionWordTest {
+    public static void main(String[] args) throws Exception {
+        for (UIManager.LookAndFeelInfo laf :
+                UIManager.getInstalledLookAndFeels()) {
+            System.out.println("Testing LAF: " + laf.getClassName());
+            SwingUtilities.invokeAndWait(() -> {
+                if (setLookAndFeel(laf)) {
+                    runTest();
+                }
+            });
+        }
+    }
+
+    private static boolean setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+            return true;
+        } catch (UnsupportedLookAndFeelException  e) {
+            System.err.println("Skipping unsupported look and feel:")
+            e.printStackTrace();
+            return false;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void runTest() {
+        String str = "one two three";
+        JPasswordField field = new JPasswordField(str);
+        if (!(field.getUI() instanceof BasicTextUI)) {
+            System.out.println("Skipping PasswordSelectionWordTest " +
+                    "because the JPasswordField UI was " + field.getUI());
+            return;
+        }
```

I'd say that this is unexpected, and it's better to throw an exception and fail the test. If not fail the test, then throw `jtreg.SkippedException` to indicate the test doesn't run.

---

OK, now we throw a RuntimeException to fail the test